base/client.jl

```diff
@@ -323,6 +323,16 @@ function exec_options(opts)
             end
         end
     end
+
+    ret = 0
+    if isdefined(Main, :main)
+        if Core.Compiler.generating_sysimg()
+            precompile(Main.main, (typeof(ARGS),))
+        else
+            ret = invokelatest(Main.main, ARGS)
+        end
```

Comment:
actually, maybe just this:
```suggestion
        end
        repl = false
```